### ISSUE TITLE
Failed to open project: failed to read files

### ISSUE THREAD

**Issue by colepatters** [Scirra/Construct-3-bugs#4951]
## Problem description

"Failed to open project. Check it is a valid Construct 3 folder project."
Errors in console: Failed to read file 'filename.webm':  TypeError: Name is not allowed.

The project used to open when I was working on it a few months ago. The last working version was r242. I do not understand why the project will not open anymore. As far as I can remember I am fairly certain the project does not contain any 3rd party addons. I even tried opening it in incognito mode so there was no chance it was any addons and it still didn't open.
## Attach a .c3p

https://drive.google.com/file/d/1t4cYsnyLeM9e7VJUvOrxcLdB90I3aSF4/view?usp=sharing
Folder project (cannot open project to save as single file)

## Steps to reproduce

1. Unzip file attached
2. Open project folder in construct 3 (any version after r242)
3. Project will not open

## Observed result

Project will not open. 

## Expected result

I expected the project to open.

## More details

**Affected browsers/platforms:** 
I tested the issue on Chrome and Edge on my Windows 11 PC and my Windows 10 laptop.

**First affected release:** 
Last working release was r242, but now that I go back to that version now it still doesn't open.

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 91.0.4472.124
Browser engine: Chromium
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 12
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36
C3 release: r240 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 286 gb
Storage usage (approx): 189 mb (0.1%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, Radeon RX 5500 XT Direct3D11 vs_5_0 ps_5_0, D3D11-27.20.11035.3002)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
This is a [bug in Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=1192050) - currently it does not allow files with the `~` character when using folder-based projects. You should be able to zip the folder and open it as a single-file project.

However in addition to that the project appears to be missing some files (probably for the same reason): when opening as a single-file project the console logs that "music\35 - wario battle ~ phase 1.webm" is missing. If you replace that file in the folder project, zip it, and then open it as a single file project, it should work. If any further errors happen, press F12 and check the browser console in case any further files are missing.

**Comment by colepatters** [comment]
Awesome, I'm pretty sure Onedrive removed all the files with the "~" character when I moved all of them to the cloud. I had to track down copies from somewhere online to replace them with, then I zipped it, opened the project, renamed the files, then saved as a folder again. 

Thank you so much for your help, reaching out to the Construct staff/community has always greeted me with a quick and helpful response.
